src/PrestaShopBundle/Tests/Form/FormFieldProviderTest.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * 2007-2017 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2017 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShopBundle\Tests\Form;
+
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Component\DependencyInjection\Container;
+
+class FormFieldProviderTest extends KernelTestCase
+{
+    /**
+     * @var Container
+     */
+    private $container;
+
+    public function setup()
+    {
+        $kernel = $this->createKernel();
+        $kernel->boot();
+        $this->container = $kernel->getContainer();
+    }
+
+    public function testTodoTypeFormIsAppended()
+    {
+        $todoForm = $this->container->get('form.factory')->create('PrestaShopBundle\Tests\Form\TodoType');
+
+        $this->assertSame(count($todoForm->all()), 4);
```

Comment:
TodoType is supposed to have only three fields, but we have registred a new one thanks to the Form Provider